src/data/real/pi/wallis.lean

```diff
@@ -5,70 +5,228 @@ Authors: Hanting Zhang
 -/
 import analysis.special_functions.integrals
 
-/-! ### The Wallis Product for Pi -/
+/-! # The Wallis formula for Pi
 
-namespace real
+This file establishes the Wallis product for `π` (`real.tendsto_prod_pi_div_two`). Our proof is
+largely about analyzing the behaviour of the sequence `∫ x in 0..π, sin x ^ n` as `n → ∞`.
+See: https://en.wikipedia.org/wiki/Wallis_product
+
+The proof can be broken down into two pieces. The first step (carried out in
+`analysis.special_functions.integrals`) is to use repeated integration by parts to obtain an
+explicit formula for this integra, which is rational if `n` is odd and a rational multiple of `π`
+if `n` is even.
+
+The second step, carried out here, is to estimate the ratio
+`∫ (x : ℝ) in 0..π, sin x ^ (2 * k + 1) / ∫ (x : ℝ) in 0..π, sin x ^ (2 * k)` and prove that
+it converges to one using the squeeze theorem. The final product for `π` is obtained after some
+algebraic manipulation. -/
```

Comment:
Can you mention in the docstring the definition `wallis.W` you introduce in the file?